Plugins/SqliteGameDB/Source/SqliteGameDB/Public/DbStatement.h

```diff
@@ -119,7 +119,8 @@ class SQLITEGAMEDB_API UDbStatement : public UObject
 	T* CreateObjectFromData()
 	{
 		T* ResultObject = NewObject<T>();
-		ReadIntoObject(ResultObject);
+		if (!ReadIntoObject(ResultObject))
+			return nullptr;
```

Comment:
@Ysgrathe so, this has been working for me pretty well so far, but I think one of the questions I have right now: do we need to do something with that `NewObject<T>()` initialized a few lines up? Will UE clean it up once there are no more references to it?

Reply:
Unreal should flag it for GC immediately we leave the function, as there will be no valid pointers to it anywhere.
Perhaps I will include a call to Destroy() on fail state, not so much to force a cleanup, (as im sure it gets cleaned up just the same anyway) but more to signify the *intent* of the code.